lib/telegraf/sidekiq.rb

```diff
@@ -37,10 +37,10 @@ module Sidekiq
     #     Only present for "normal" (async) jobs (with tag `type` of "job").
     #
     class Middleware
-      def initialize(agent:, series: 'sidekiq', tags: {})
-        @agent = agent
-        @series = series.to_s.freeze
-        @tags = tags.freeze
+      def initialize(options)
+        @agent = options.fetch(:agent)
+        @series = options.fetch(:series, 'sidekiq').to_s.freeze
+        @tags = options.fetch(:tags) { {} }.freeze
```

Comment:
I'd prefer having `agent` as a required position argument and optional options. Looks more like a usual non-kwargs signature to me.

```suggestion
      def initialize(agent, options = {})
        @agent = agent
        @series = options.fetch(:series, 'sidekiq').to_s.freeze
        @tags = options.fetch(:tags, {}).freeze
```

Reply:
Maybe you should take over, we can just leave this as an open issue there there's a Ruby 2.7 dep